app/models/node.js

```diff
@@ -122,16 +122,11 @@ var Node = Resource.extend(StateCounts, ResourceUsage, {
     const roles = get(this, 'rolesArray');
 
     if ( roles.length >= 3 ) {
-      return [intl.t('generic.all')];
```

Comment:
I think this just needed to watch `intl.locale`, which I added separately... If the roles are not the actual translated strings then searching by role becomes weird/broken.